New errors:
```
/home/runner/work/quokka/quokka/src/ODEIntegrate.hpp:34:13: error: SYCL kernel cannot call through a function pointer
   34 |         int ierr = rhs(t0, y_arg, k1, user_data);
      |                    ^
/home/runner/work/quokka/quokka/src/ODEIntegrate.hpp:43:9: error: SYCL kernel cannot call through a function pointer
   43 |         ierr = rhs(t0 + dt, y_arg, k2, user_data);
      |                ^
/home/runner/work/quokka/quokka/src/ODEIntegrate.hpp:136:2: error: SYCL kernel cannot call through a function pointer
  136 |         rhs(t0, y0, ydot0, user_data);
      |         ^
```

```
In file included from /home/runner/work/quokka/quokka/src/EOS.hpp:20:
/home/runner/work/quokka/quokka/extern/Microphysics/interfaces/eos.H:86:35: error: SYCL kernel cannot use a non-const global variable
   86 |   state.rho = amrex::min(EOSData::maxdens, amrex::max(EOSData::mindens, state.rho));
      |                                   ^
/home/runner/work/quokka/quokka/extern/Microphysics/interfaces/eos.H:86:64: error: SYCL kernel cannot use a non-const global variable
   86 |   state.rho = amrex::min(EOSData::maxdens, amrex::max(EOSData::mindens, state.rho));
      |                                                                ^
/home/runner/work/quokka/quokka/extern/Microphysics/interfaces/eos.H:94:33: error: SYCL kernel cannot use a non-const global variable
   94 |   state.T = amrex::min(EOSData::maxtemp, amrex::max(EOSData::mintemp, state.T));
      |                                 ^
/home/runner/work/quokka/quokka/extern/Microphysics/interfaces/eos.H:94:62: error: SYCL kernel cannot use a non-const global variable
   94 |   state.T = amrex::min(EOSData::maxtemp, amrex::max(EOSData::mintemp, state.T));
      |                                                              ^
/home/runner/work/quokka/quokka/extern/Microphysics/interfaces/eos.H:86:35: error: SYCL kernel cannot use a non-const global variable
   86 |   state.rho = amrex::min(EOSData::maxdens, amrex::max(EOSData::mindens, state.rho));
      |                                   ^
/home/runner/work/quokka/quokka/extern/Microphysics/interfaces/eos.H:86:64: error: SYCL kernel cannot use a non-const global variable
   86 |   state.rho = amrex::min(EOSData::maxdens, amrex::max(EOSData::mindens, state.rho));
      |                                                                ^
/home/runner/work/quokka/quokka/extern/Microphysics/interfaces/eos.H:113:32: error: SYCL kernel cannot use a non-const global variable
  113 |         if (state.h < EOSData::minh || state.h > EOSData::maxh) {
      |                                ^
```

--- a/.github/workflows/intel.yml
+++ b/.github/workflows/intel.yml
@@ -40,7 +40,6 @@ jobs:
         set -e
         cmake -S . -B build                                \
             -DCMAKE_VERBOSE_MAKEFILE=ON                    \
-            -DAMReX_GPU_BACKEND=SYCL                       \
             -DCMAKE_C_COMPILER=$(which icx)                \
             -DCMAKE_CXX_COMPILER=$(which icpx)             \
             -DCMAKE_CXX_COMPILER_LAUNCHER=ccache           \

--- a/.github/workflows/intel.yml
+++ b/.github/workflows/intel.yml
@@ -14,7 +14,6 @@ jobs:
     - uses: actions/checkout@v4
       with:
         submodules: true
-
     - name: Dependencies
       run: |
         .github/workflows/dependencies/dependencies_dpcpp.sh
@@ -49,7 +48,17 @@ jobs:
 
         ccache -s
         du -hs ~/.cache/ccache
-
+    - name: Test
+      working-directory: ${{runner.workspace}}/build
+      shell: bash
+      run: ctest --output-on-failure -C $BUILD_TYPE
+    - name: Upload test output
+      if: always()
+      uses: actions/upload-artifact@v4
+      with:
+        name: test-results
+        path: ${{github.workspace}}/tests
+        
   save_pr_number:
     if: github.event_name == 'pull_request'
     runs-on: ubuntu-latest

--- a/.github/workflows/intel.yml
+++ b/.github/workflows/intel.yml
@@ -14,10 +14,12 @@ jobs:
     - uses: actions/checkout@v4
       with:
         submodules: true
+        
     - name: Dependencies
       run: |
         .github/workflows/dependencies/dependencies_dpcpp.sh
         .github/workflows/dependencies/dependencies_ccache.sh
+        
     - name: Set Up Cache
       uses: actions/cache@v4
       with:
@@ -25,6 +27,7 @@ jobs:
         key: ccache-${{ github.workflow }}-${{ github.job }}-git-${{ github.sha }}
         restore-keys: |
              ccache-${{ github.workflow }}-${{ github.job }}-git-
+             
     - name: Build & Install
       env: {CXXFLAGS: "-fno-operator-names -Werror -Wall -Wextra -Wpedantic -Wnull-dereference -Wfloat-conversion -Wshadow -Woverloaded-virtual -Wextra-semi -Wunreachable-code -Wnon-virtual-dtor"}
       run: |
@@ -48,10 +51,15 @@ jobs:
 
         ccache -s
         du -hs ~/.cache/ccache
-    - name: Test
+        
+    - name: Create output directory
+      run: cmake -E make_directory $GITHUB_WORKSPACE/tests
+      
+    - name: Run tests
       working-directory: ${{runner.workspace}}/build
       shell: bash
       run: ctest --output-on-failure -C $BUILD_TYPE
+      
     - name: Upload test output
       if: always()
       uses: actions/upload-artifact@v4

